### DISCUSSION TITLE
Could we drag scripts to Scene to add a new Node?

### DISCUSSION THREAD

**Discussion by theraot** [godotengine/godot-proposals#7211] [Editor]
I've found myself wanting to drag a script from the FileSystem to the Scene to add a new Node of the appropriate type for the script and with it attached.

However, dragging a script to the Scene attaches the script to an existing Node, which is not what I want.

Thus, I imagine that it could be implemented with key combination (e.g. holding Ctrl).

And if a Script that does not extend a Node type is dragged to the Scene, it should fail, preferably with an error message.

*Currently it is possible to attach a Resource script to a Node. But running an scene where such script is attached to a Node will fail.*

**Comment by passivestar** [comment]
Got the same idea, down to the modifier key. Dragging multiple scripts could create multiple child nodes. Curious to hear what more people think about this

**Comment by Gnumaru** [comment]
hey, this is already possible. just ctrl+drag and drop a script.

**Reply by passivestar** [reply]
oh wow, how could I miss that. I guess I was instinctively trying cmd on mac instead or ctrl. It doesn't work with multiple scripts btw, but not sure if it's gonna be a big deal yet

**Comment by theraot** [comment]
If anything, the fact that I didn't know the feature existed, opened a discussion, people passed by, upvoted, and nobody told me until now... It tells me that this is not easy to discover. So I have posted about it here: https://github.com/godotengine/godot-docs/issues/4377

I'm closing the discussion.

**Comment by gnat** [comment]
Strongly feel this should be default behavior... would definitely encourage more composition.

_Definitely needs a tooltip popup explaining `CTRL` when dragging a script._
